src/main/java/jp/co/yahoo/yosegi/binary/maker/OptimizedNullArrayDumpDoubleColumnBinaryMaker.java

```diff
@@ -184,19 +189,148 @@ public int calcBinarySize( final IColumnAnalizeResult analizeResult ) {
+
+    ICompressor compressor = FindCompressor.get(columnBinary.compressorClassName);
+    byte[] binary = compressor.decompress(columnBinary.binary, start, length);
+
+    ByteBuffer wrapBuffer = ByteBuffer.wrap(binary, 0, binary.length);
 
-    return new LazyColumn(
-      columnBinary.columnName ,
-      columnBinary.columnType ,
-      new ColumnManager(
-        columnBinary
-      )
-    );
+    ByteOrder order = wrapBuffer.get() == (byte) 0 ? ByteOrder.BIG_ENDIAN : ByteOrder.LITTLE_ENDIAN;
+    int startIndex = wrapBuffer.getInt();
+    int nullIndexLength = wrapBuffer.getInt();
+    int valueBinaryLength = binary.length - META_LENGTH - nullIndexLength;
+
+    boolean[] isNullArray = NullBinaryEncoder.toIsNullArray(binary, META_LENGTH, nullIndexLength);
+
+    IReadSupporter valueReader =
+        ByteBufferSupporterFactory.createReadSupporter(
+            binary, META_LENGTH + nullIndexLength, valueBinaryLength, order);
+
+    // NOTE: Calculate dictionarySize
+    int dictionarySize = 0;
+    int previousLoadIndex = -1;
+    int lastIndex = startIndex + isNullArray.length - 1;
+    for (int loadIndex : columnBinary.loadIndex) {
+      if (loadIndex < 0) {
+        throw new IOException("Index must be equal to or greater than 0.");
+      } else if (loadIndex < previousLoadIndex) {
+        throw new IOException("Index must be equal to or greater than the previous number.");
+      }
+      if (loadIndex > lastIndex) {
+        break;
+      }
+      if (loadIndex >= startIndex && !isNullArray[loadIndex - startIndex]) {
+        if (previousLoadIndex != loadIndex) {
+          dictionarySize++;
+        }
+      }
+      previousLoadIndex = loadIndex;
+    }
+    loader.createDictionary(dictionarySize);
+
+    // NOTE:
+    //   Set value to dict: dictionaryIndex, value
+    //   Set dictionaryIndex: loadIndexArrayOffset, dictionaryIndex
+    previousLoadIndex = -1; // NOTE: reset
+    int loadIndexArrayOffset = 0;
+    int dictionaryIndex = -1;
+    int readOffset = startIndex;
+    double value = 0;
+    for (int loadIndex : columnBinary.loadIndex) {
+      if (loadIndex > lastIndex) {
+        break;
+      }
+      // NOTE: read skip
+      for (; readOffset <= loadIndex; readOffset++) {
+        if (readOffset >= startIndex && !isNullArray[readOffset - startIndex]) {
+          value = valueReader.getDouble();
+        }
+      }
+      if (loadIndex < startIndex || isNullArray[loadIndex - startIndex]) {
+        loader.setNull(loadIndexArrayOffset);
+      } else {
+        if (previousLoadIndex != loadIndex) {
+          dictionaryIndex++;
+          loader.setDoubleToDic(dictionaryIndex, value);
+        }
+        loader.setDictionaryIndex(loadIndexArrayOffset, dictionaryIndex);
+      }
+      previousLoadIndex = loadIndex;
+      loadIndexArrayOffset++;
+    }
+
+    // NOTE: null padding for invalid loadIndexes
+    for (int i = loadIndexArrayOffset; i < loader.getLoadSize(); i++) {
+      loader.setNull(i);
+    }
```

Comment:
After this process, please add a process to set NULL from startIndex + isNullArray.length to loader.getLoadSize ().

Reply:
I fixed the points that were pointed out.